Standard/src/Diagnostics/Emulation/Internal.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using Microsoft.Quantum.Diagnostics.Emulation;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+using NumSharp;
+using static NumSharp.Slice;
+
+namespace Microsoft.Quantum.Diagnostics
+{
+    internal class ArrayDumper : QuantumSimulator.StateDumper
+    {
+        // NB: NumSharp does not yet support complex numbers, so we store data
+        //     as an array with a trailing index of length 2.
+        internal NumSharp.NDArray? Data = null;
+        public ArrayDumper(QuantumSimulator sim) : base(sim)
+        {
+
+        }
+
+        public override bool Callback(uint idx, double real, double img)
+        {
+            if (Data as object == null) throw new Exception("Expected data buffer to be initialized before callback, but it was null.");
+            Data[(int)idx, 0] = real;
+            Data[(int)idx, 1] = img;
+            return true;
+        }
+
+        public override bool Dump(IQArray<Qubit>? qubits = null)
+        {
+            var count = qubits == null
+                        ? this.Simulator.QubitManager!.GetAllocatedQubitsCount()
+                        : qubits.Length;
+            var nQubitsPerRegister = ((int) count / 2);
```

Comment:
```suggestion
            var nQubitsPerRegister = ((int)count / 2);
```

Seems that `count` is cast, but it looks like `count / 2` is cast because of the space.